services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/monitor/task.py

```diff
@@ -0,0 +1,372 @@
+import asyncio
+import logging
```

Comment:
this sub module ``dynamic_sidecar/monitor`` is probably the most important part and there is **zero** test coverage :-( !

Reply:
Please run the unit tests with coverage enabled. Before asking for review I've made sure the most important paths were all covered inside the entire `dynamic_sidecar` module.

```
----------- coverage: platform linux, python 3.8.7-final-0 -----------
Name                                                                          Stmts   Miss  Cover   Missing
-----------------------------------------------------------------------------------------------------------
src/simcore_service_director_v2/modules/dynamic_sidecar/__init__.py               1      0   100%
src/simcore_service_director_v2/modules/dynamic_sidecar/client_api.py            68      7    90%   65, 89-91, 99-104
src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api.py           146      9    94%   75, 89, 99, 147, 170-171, 200, 223, 237
src/simcore_service_director_v2/modules/dynamic_sidecar/docker_states.py         60     11    82%   42, 46, 48, 51-52, 61, 63, 65, 67, 69, 71
src/simcore_service_director_v2/modules/dynamic_sidecar/errors.py                13      0   100%
src/simcore_service_director_v2/modules/dynamic_sidecar/module_setup.py          12      0   100%
src/simcore_service_director_v2/modules/dynamic_sidecar/monitor/__init__.py       1      0   100%
src/simcore_service_director_v2/modules/dynamic_sidecar/monitor/abc.py           10      0   100%
src/simcore_service_director_v2/modules/dynamic_sidecar/monitor/events.py         3      0   100%
src/simcore_service_director_v2/modules/dynamic_sidecar/monitor/task.py         153      8    95%   87-88, 97, 155, 338-342, 363-364
-----------------------------------------------------------------------------------------------------------
TOTAL                                                                           467     35    93%
```

Reply:
Trying to get the CI green. My last commit should do it.

Reply:
You raised a good point: i was wrong, it has **very high** test coverage and I will check the coverage next time 🤕 

Nonetheless, when I wrote that I was  thinking on specific unit-tests per module. When I create a brand new module and want to know how it is used, I expect an associated unit test, e.g. if I would like to know about how to use e.g. monitor, i would have a ``test_modules_dynamic_sidecar_monitor.py`` where I have some real unit-tests.

In this case, coverage comes through other paths (still valid) but still do not fulfill my view of what a unit-tests is.